.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,152 @@
+### Contributing to ZFS on Linux
+
+<p align="center"><img src="http://zfsonlinux.org/images/zfs-linux.png"/></p>
+
+*First of all, thank you for taking the time to contribute!*
+
+By using the following guidelines, you can help us make ZFS on Linux even better.
+
+#### Table Of Contents
+
+[What should I know before I get started?](#what-should-i-know-before-i-get-started)
+
+  * [Get ZFS](#get-zfs)
+  * [Debug ZFS](#debug-zfs)
+  * [Where can I ask for help?](#where-can-I-ask-for-help)
+  
+[How Can I Contribute?](#how-can-i-contribute)
+
+  * [Reporting Bugs](#reporting-bugs)
+  * [Suggesting Enhancements](#suggesting-enhancements)
+  * [Pull Requests](#pull-requests)
+  * [Testing](#testing)
+  
+[Style Guides](#style-guides)
+ 
+  * [Coding Conventions](#coding-conventions)
+
+Helpful resources
+
+  * [ZFS on Linux wiki](https://github.com/zfsonlinux/zfs/wiki)
+  * [OpenZFS Documentation](http://open-zfs.org/wiki/Developer_resources)
+
+
+## What should I know before I get started?
+
+
+### Get ZFS
+
+You can build zfs packages by following 
+[these instructions](https://github.com/zfsonlinux/zfs/wiki/Building-ZFS), 
+or install stable packages from 
+[your distribution's repository](https://github.com/zfsonlinux/zfs/wiki/Getting-Started).
+
+
+### Debug ZFS
+
+A variety of methods and tools are available to aid ZFS developers.  It's strongly recommended
+that when developing a patch the `--enable-debug` configure option should be set.
+This will enable additional correctness checks and all the ASSERTs to help quickly catch potential issues.
```

Comment:
Since this markdown file will live in the repository please make sure all the lines are wrapped at 80 characters so it'll be readable as plain text.